### PR TITLE
tools/rpki-rov: improve input validation

### PR DESCRIPTION
**Pull request by mroethke**
### Contribution description

The input validation was insufficient which could lead to buffer
overflows, uninitialized jumps and undefined behaviour.

coverity: 1477248

### Testing procedure
Use rpki-rov and enter a line like "999999999999 e f", it will happily accept and try to validate it.

### CODE REVIEW

File: tools/rpki-rov.c [hunk 1]
```diff
@@ -43,6 +45,18 @@ int connection_error(enum rtr_mgr_status status)
 	return 0;
 }
 
+static int str_to_int(const char *str, int *value) {
```
Comment (smlng) [inline tools/rpki-rov.c]: bracket on new line

File: tools/rpki-rov.c [hunk 2]
```diff
@@ -43,6 +45,18 @@ int connection_error(enum rtr_mgr_status status)
...
+	errno = 0;
+	int tmp = strtol(str, NULL, 10);
+
+	if (errno == 0) {
```
Comment (smlng) [inline tools/rpki-rov.c]: recently found a nice post on how to write clean and maintainable `if`s that mainly suggested to rather check for negatives (errors) not positives (as you btw. did below already). Meaning: rewrite this to 
```
if (errno != 0)
        return 1;

*value = tmp;
return 0;
```

File: tools/rpki-rov.c [hunk 3]
```diff
@@ -141,17 +152,46 @@ int main(int argc, char *argv[])
 		char *input_tok = NULL;
 
 		input_tok = strtok(input, delims);
-		strcpy(ip, input_tok);
+		struct lrtr_ip_addr pref;
+		char ip[INET6_ADDRSTRLEN];
+
+		if (strlen(input_tok) > sizeof(ip) - 1) {
+			printf("Error: Invalid ip addr\n");
```
Comment (smlng) [inline tools/rpki-rov.c]: maybe use `fprintf(stderr, ...)`  here and below?

Comment (mroethke) [reply tools/rpki-rov.c]: good idea, will also remove the flush calls, as they are not necessary with stderr.

File: tools/rpki-rov.c (hunk 2 above)
Comment (mroethke) [reply tools/rpki-rov.c]: Done.
Do you have a link? Would love to read more about that.

File: tools/rpki-rov.c (hunk 1 above)
Comment (mroethke) [reply tools/rpki-rov.c]: done

File: tools/rpki-rov.c (hunk 2 above)
Comment (smlng) [reply tools/rpki-rov.c]: actually was a youtube video: https://www.youtube.com/watch?v=ldqDpmMkXgw

**Comment by smlng** [comment]
please squash

**Pull request merged**